doschema/jsvalidation.py

```diff
@@ -0,0 +1,316 @@
+                'dependencies' not in schema):
+            err_msg = "Root field / type is not 'object' in {0}"
+            raise JSONSchemaCompatibilityException(
+                '/',
+                self.uri,
+                err_msg.format(self.uri)
+            )
+
+        self.validate_schema(schema, ())
+
+        for path, field in six.iteritems(self.fields_types_dict):
+            if field.field_type == "null":
+                logging.warning(
+                    'Null type in field ' + self.make_json_pointer(
+                            field.field_tuple
+                        )
+                )
+
+        self.uri = None
+        self.resolver = None
+
+    def validate_schema(self, curr_schema, curr_field):
+        """Go through the schema and retrieve schema's particular fields.
+
+        :param curr_schema: Schema or subschema that is currently processed.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        while '$ref' in curr_schema:
+            path = curr_schema.get('$ref')
+            curr_schema = self.resolver.resolve(path)[1]
+
+        self.validate_collection(curr_schema, curr_field)
+
+        field = self.fields_types_dict.get(curr_field)
+
+        if "type" in curr_schema:
+            if field:
+                if field.field_type == "null":
+                    logging.warning(
+                        "There was overriden type 'null' in {0} in schema {1}".
+                        format(
+                            self.make_json_pointer(curr_field),
+                            self.uri
+                        )
+                    )
+                elif curr_schema['type'] != field.field_type:
+                    err_msg = "{0} type mismatch in schemas {1} and {2}."
+                    raise JSONSchemaCompatibilityException(
+                        field.schema_index,
+                        self.uri,
+                        err_msg.format(
+                            self.make_json_pointer(curr_field),
+                            field.schema_index,
+                            self.uri
+                        )
+                    )
+            self.fields_types_dict[curr_field] = FieldToAdd(
+                schema_index=self.uri,
+                field_tuple=curr_field,
+                field_type=curr_schema['type']
+            )
+
+            if curr_schema['type'] == 'array':
+                self.validate_array(
+                    curr_schema['items'],
+                    curr_field + ('items',)
+                )
+        if "enum" in curr_schema:
+            enum_type = self.validate_enum(
+                curr_schema['enum'],
+                curr_field,
+                curr_schema.get('type')
+            )
+            if field is not None and field.field_type != enum_type:
+                err_msg = "Enum {0} conflicting itself error in schema {1}."
+                raise JSONSchemaEnumException(
```

Comment:
This is a type conflict so it might be better to use the same exception, i.e. `JSONSchemaCompatibilityException `